commit d292dd0eb3ac6ce6ea66715bb9f6b8e2ae70747c
Author: Qiuxu Zhuo <qiuxu.zhuo@intel.com>
Date:   Fri Nov 20 16:10:36 2020 -0800

    PCI/AER: Add RCEC AER error injection support
    
    Root Complex Event Collectors (RCEC) appear as peers to Root Ports and may
    also have the AER capability.
    
    Add RCEC support to the AER error injection driver.
    
    Co-developed-by: Sean V Kelley <sean.v.kelley@intel.com>
    Link: https://lore.kernel.org/r/20201121001036.8560-16-sean.v.kelley@intel.com
    Tested-by: Jonathan Cameron <Jonathan.Cameron@huawei.com> # non-native/no RCEC
    Signed-off-by: Qiuxu Zhuo <qiuxu.zhuo@intel.com>
    Signed-off-by: Sean V Kelley <sean.v.kelley@intel.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Reviewed-by: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@linux.intel.com>

diff --git a/drivers/pci/pcie/aer_inject.c b/drivers/pci/pcie/aer_inject.c
index c2cbf425afc5..767f8859b99b 100644
--- a/drivers/pci/pcie/aer_inject.c
+++ b/drivers/pci/pcie/aer_inject.c
@@ -333,8 +333,11 @@ static int aer_inject(struct aer_error_inj *einj)
 	if (!dev)
 		return -ENODEV;
 	rpdev = pcie_find_root_port(dev);
+	/* If Root Port not found, try to find an RCEC */
+	if (!rpdev)
+		rpdev = dev->rcec;
 	if (!rpdev) {
-		pci_err(dev, "Root port not found\n");
+		pci_err(dev, "Neither Root Port nor RCEC found\n");
 		ret = -ENODEV;
 		goto out_put;
 	}